modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Alternative.kt

```diff
@@ -1,6 +1,16 @@
 package arrow.typeclasses
 
+import arrow.Kind
+import arrow.core.ListK
+import arrow.core.k
+
 /**
  * ank_macro_hierarchy(arrow.typeclasses.Alternative)
  */
-interface Alternative<F> : Applicative<F>, MonoidK<F>
+interface Alternative<F> : Applicative<F>, MonoidK<F> {
+  fun <A> Kind<F, A>.some(): Kind<F, ListK<A>> = map(this, many()) { (v, vs) -> (listOf(v) + vs).k() }
+
+  fun <A> Kind<F, A>.many(): Kind<F, ListK<A>> = some() alt just(emptyList<A>().k())
+
+  infix fun <A> Kind<F, A>.alt(b: Kind<F, A>): Kind<F, A>
```

Comment:
Please provide at least one non-infix alternative just in case :P

Reply:
Should I leave alt as the infix version and provide an orElse non-infix function? I wish we could use operators and name it <|> 😅

Reply:
ye, works for me

Reply:
One nit. Could you please swap things around so `alt` is implemented in terms of `orElse` instead?

Reply:
Sure 👍